#### 1. Take a number and find the sum of its digits. Example: 1234 → 10

In [2]:
num = int(input("Enter a number: "))
sum_digits = 0
while num > 0:
    digit = num % 10
    sum_digits += digit
    num = num // 10
print("Sum of digits:", sum_digits)

Enter a number:  1234


Sum of digits: 10


#### 2. Take a number and count how many digits it contains.

In [3]:
num = int(input("Enter a number: "))
count = 0
while num > 0:
    num = num // 10
    count += 1
print("Number of digits:", count)

Enter a number:  12345


Number of digits: 5


#### 3. Reverse a number without converting it to a string. Example: 12345 → 54321

In [4]:
num = int(input("Enter a number: "))
reverse = 0
while num > 0:
    digit = num % 10
    reverse = reverse * 10 + digit
    num = num // 10
print("Reversed number:", reverse)

Enter a number:  123456


Reversed number: 654321


#### 4. Check whether a number is an Armstrong number. Example: 153 → Armstrong


In [6]:
num = int(input("Enter a number: "))
original = num
digits = 0
temp = num
while temp > 0:
    temp = temp // 10
    digits += 1

sum_digits = 0
temp = num
while temp > 0:
    digit = temp % 10
    sum_digits += digit ** digits
    temp = temp // 10
if sum_digits == original:
    print("Armstrong")
else:
    print("Not Armstrong")

Enter a number:  153


Armstrong


#### 5. Print the first n terms of the Fibonacci series. Example: 0 1 1 2 3 5 8

In [7]:
n = int(input("Enter the number of terms: "))
a = 0
b = 1
for i in range(n):
    print(a, end=" ")
    a, b = b, a + b

Enter the number of terms:  7


0 1 1 2 3 5 8 

#### 6. Take a number and print its multiplication table from 1 to 10.

In [8]:
num = int(input("Enter a number: "))
for i in range(1, 11):
    print(num, "x", i, "=", num * i)

Enter a number:  5


5 x 1 = 5
5 x 2 = 10
5 x 3 = 15
5 x 4 = 20
5 x 5 = 25
5 x 6 = 30
5 x 7 = 35
5 x 8 = 40
5 x 9 = 45
5 x 10 = 50


#### 7. Check whether a number is a perfect number. Example: 6 → 1 + 2 + 3 = 6

In [9]:
num = int(input("Enter a number: "))
sum_divisors = 0
for i in range(1, num):
    if num % i == 0:
        sum_divisors += i
if sum_divisors == num:
    print("Perfect Number")
else:
    print("Not a Perfect Number")

Enter a number:  6


Perfect Number


#### 8. Take a number and print all of its factors.

In [10]:
num = int(input("Enter a number: "))
print("Factors are:")
for i in range(1, num + 1):
    if num % i == 0:
        print(i, end=" ")

Enter a number:  12


Factors are:
1 2 3 4 6 12 

#### 9. Take two numbers and print all prime numbers between them.

In [11]:
start = int(input("Enter the first number: "))
end = int(input("Enter the second number: "))
print("Prime numbers:")
for num in range(start, end + 1):
    if num < 2:
        continue
    is_prime = True
    for i in range(2, num):
        if num % i == 0:
            is_prime = False
            break
    if is_prime:
        print(num, end=" ")

Enter the first number:  10
Enter the second number:  30


Prime numbers:
11 13 17 19 23 29 

#### 10. Find the Greatest Common Divisor (GCD) of two numbers without using math.gcd().

In [12]:
num1 = int(input("Enter first number: "))
num2 = int(input("Enter second number: "))
while num2 != 0:
    num1, num2 = num2, num1 % num2
print("GCD:", num1)

Enter first number:  24
Enter second number:  36


GCD: 12
